frontend/src/concepts/analyticsTracking/useSegmentTracking.ts

```diff
@@ -10,28 +11,18 @@ export const useSegmentTracking = (): void => {
   const { dashboardConfig } = useAppContext();
   const username = useAppSelector((state) => state.user);
   const clusterID = useAppSelector((state) => state.clusterID);
+  const userProps = useTrackUser(username);
 
   React.useEffect(() => {
     if (segmentKey && loaded && !loadError && username && clusterID) {
-      const computeUserId = async () => {
-        const anonymousIDBuffer = await crypto.subtle.digest(
-          'SHA-1',
-          new TextEncoder().encode(username),
-        );
-        const anonymousIDArray = Array.from(new Uint8Array(anonymousIDBuffer));
-        return anonymousIDArray.map((b) => b.toString(16).padStart(2, '0')).join('');
-      };
-
       window.clusterID = clusterID;
       initSegment({
         segmentKey,
         enabled: !dashboardConfig.spec.dashboardConfig.disableTracking,
       }).then(() => {
-        computeUserId().then((userId) => {
-          fireIdentifyEvent({ anonymousID: userId });
-          firePageEvent();
-        });
+        fireIdentifyEvent(userProps);
```

Comment:
It seems possible that `userProps.anonymousId` may not yet be defined because it is computed asynchronously.

Reply:
This comment still holds true. 
Perhaps you should capture a value to know when segment is initialized, then use a `useEffect` on that value along with `userProps` so that you fire an identify event whenever 'userProps.anonymousID' changes?

The old implementation didn't suffer from this issue.